### ISSUE TITLE
The tooltip for a theory shows a GUID instead of an argument's name and ToString() value

### ISSUE THREAD

**Issue by stijnherreman** [xunit/xunit#1454]
In the Test Explorer, a theory is displayed as `Foo.Bar(baz: bat)`. A test name is often long enough to be cut off, so I hover the test name to see the tooltip. The tooltip doesn't show the argument's name and `ToString()` value, instead it shows (for example) `Foo.Bar (301a52059c85406a9716a3d51f87f238)`.

Can the tooltip be changed to show the same information?

Example test:

![xunit theory tooltip](https://user-images.githubusercontent.com/1618054/30441333-69d9b20e-9979-11e7-9989-920677515f7b.png)

```c#
[Theory]
[MemberData(nameof(FooTestCases))]
public void Foo(MemberDataWrapper<TestCase> testCase)
{
}

public class TestCase
{
}

private static IEnumerable<object[]> FooTestCases()
{
    yield return new object[]
    {
        new MemberDataWrapper<TestCase>("case1", new TestCase())
    };

    yield return new object[]
    {
        new MemberDataWrapper<TestCase>("case2", new TestCase())
    };
}
```

Source for the `MemberDataWrapper` class:

```c#
using Newtonsoft.Json;
using Xunit.Abstractions;

namespace MyNamespace
{
    /*
     * Inspired by https://stackoverflow.com/a/43068778
     */
    public sealed class MemberDataWrapper<T>
        : IXunitSerializable
    {
        public string Description { get; private set; }

        public T Value { get; private set; }

        public MemberDataWrapper()
        {
        }

        public MemberDataWrapper(string description, T value)
        {
            Description = description;
            Value = value;
        }

        public void Deserialize(IXunitSerializationInfo info)
        {
            Description = info.GetValue<string>(nameof(Description));
            Value = JsonConvert.DeserializeObject<T>(info.GetValue<string>(nameof(Value)));
        }

        public void Serialize(IXunitSerializationInfo info)
        {
            info.AddValue(nameof(Description), Description);
            info.AddValue(nameof(Value), JsonConvert.SerializeObject(Value));
        }

        public override string ToString()
        {
            return Description;
        }
    }
}
```

**Comment by bradwilson** [comment]
I believe this is an issue that has to be fixed by Visual Studio, not us. What you're seeing is two different values: the *display name* (the bit that's being cut off), and then VS is showing you the *unique ID* in the tooltip. Ideally, the tooltip should be showing the display name. Unfortunately, we can't use the display name as the unique ID, because there is simply no way for us to ensure that it's actually unique (the two values serve different purposes).

**Comment by stijnherreman** [comment]
Is it possible to make the *unique ID* a combination of a GUID and what is shown in the *display name*? That would guarantee it still being unique.

**Comment by bradwilson** [comment]
That just feels like we're hacking a fix for their bug. :(

**Comment by stijnherreman** [comment]
I just feel that getting this changed/fixed in Visual Studio has as much chance as winning the lottery.

But it would be a hack indeed, and I fully understand if you'd rather not implement such a thing. At the end of the day it's not a blocking issue, just a minor annoyance.

**Comment by sharwell** [comment]
@stijnherreman Thanks for that report. Will help put pressure on Microsoft/vstest#822 :smile:

**Comment by bradwilson** [comment]
I had a discussion with the VSTest team yesterday, and I think we can remove the GUID from the fully qualified name.

**Comment by bradwilson** [comment]
Fixed in 2a04590d

**Comment by stijnherreman** [comment]
I've received feedback on the Developer Community report:

**John Fleisher [MSFT]**
> Thanks for this feedback. Your use-case is a common one. We will investigate a solution that will preserve the ability to discern the namespace, class, and method; but improve the experience for theory tests identifiers.
